Utils/azure-explorer-common/src/com/microsoft/azure/toolkit/lib/appservice/file/AppServiceFileService.java

```diff
@@ -0,0 +1,83 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.toolkit.lib.appservice.file;
+
+import com.google.common.base.Joiner;
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.azure.management.appservice.implementation.AppServiceManager;
+import com.microsoft.rest.RestClient;
+import lombok.SneakyThrows;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class AppServiceFileService {
+
+    private final AppServiceFileClient client;
+    private final WebAppBase app;
+
+    private AppServiceFileService(final WebAppBase app, AppServiceFileClient client) {
+        this.app = app;
+        this.client = client;
+    }
+
+    public static AppServiceFileService forApp(final WebAppBase app) {
+        final AppServiceFileClient client = getClient(app);
+        return new AppServiceFileService(app, client);
+    }
+
+    public List<? extends AppServiceFile> getFilesInDirectory(String path) {
+        return this.client.getFilesInDirectory(path).toBlocking().first().stream()
+                          .filter(file -> !"text/xml".equals(file.getMime()) || !file.getName().contains("LogFiles-kudu-trace_pending.xml"))
+                          .collect(Collectors.toList());
+    }
+
+    private static AppServiceFileClient getClient(WebAppBase webAppBase) {
+        if (webAppBase.defaultHostName() == null) {
+            throw new UnsupportedOperationException("Cannot initialize kudu vfs client before web app is created");
+        } else {
+            String host = webAppBase.defaultHostName().toLowerCase().replace("http://", "").replace("https://", "");
+            final String[] parts = host.split("\\.", 2);
+            host = Joiner.on('.').join(parts[0], "scm", parts[1]);
+            final AppServiceManager manager = webAppBase.manager();
+            final RestClient restClient = getRestClient(manager);
+            return restClient.newBuilder()
+                             .withBaseUrl("https://" + host)
+                             .withConnectionTimeout(3L, TimeUnit.MINUTES)
+                             .withReadTimeout(3L, TimeUnit.MINUTES)
+                             .build()
+                             .retrofit()
+                             .create(KuduFileClient.class);
```

Comment:
AppServiceFileService should not refer the implements of `AppServiceFileClient`

Reply:
IMO,  `client` lies at the same layer as `repository`/`dao`, which are used by `service`. why do you think it should not be used in `AppServiceFileService`?

Reply:
Sorry for the confusion, I just doubt why `getClient` will create an `AppServiceFileClient` with `KuduFileClient.class`, which is the implements of ``AppServiceFileClient`, but it also make sense if we make FileClient as the strategy and difference File Service instance with different file client